### PR TITLE
ci(PR): gather pull request actions in `PR.yml`

### LINKED ISSUE #3768: Call docs and typos checks in PR.yml
### File Location(s)

https://github.com/gear-tech/gear/blob/master/.github/workflows/docs.yml

### Proposal

- move the content of `typos.yml` to `PR.yml`
- make docs a workflow call and redirect it to `PR.yml`

after these, we can gather all of the workflows in to 1 namespace, make everything cleaner

### PR DESCRIPTION
**Pull request by clearloop**
Resolves #3768

after this PR and #3767, we'll only have one name space `PR` in the checks of pull requests

- [x] rename `typos.yml`
- [x] call docs inside of `PR.yml`

@gear-tech/dev

### CODE REVIEW

**Review by StackOverflowExcept1on — APPROVED** [review]

File: .github/actions/message/action.yml [hunk 1]
```diff
@@ -12,6 +12,8 @@ outputs:
     description: If enable cache.
   check:
     description: If trigger check.
+  docs:
```
Comment (breathx) [inline .github/actions/message/action.yml:15]: docs always should be triggered I guess

**Review by breathx — APPROVED** [review]
Same as #3767  needs master fetch

File: .github/workflows/PR.yml [hunk 2]
```diff
@@ -52,6 +52,8 @@ jobs:
     steps:
       - uses: actions/checkout@v4
       - uses: ./.github/actions/docs
+        with:
+          github_token: ${{ secrets.GITHUB_TOKEN }}
```
Comment (StackOverflowExcept1on) [inline .github/workflows/PR.yml:56]: iirc secrets starting with "GITHUB_" are not supported

Comment (clearloop) [reply .github/workflows/PR.yml:56]: > iirc secrets starting with "GITHUB_" are not supported


what do you mean ?

https://docs.github.com/en/actions/security-guides/automatic-token-authentication#about-the-github_token-secret

and it works in fc33a96 as well

**Comment by breathx** [comment]
@clearloop ready to merge?

**Pull request merged**